cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericAggregator.java

```diff
@@ -73,225 +49,282 @@
-            Iterator<String> it = aggregation.keySet().iterator();
-            while (it.hasNext()) {
-                String entry = (String) it.next();
-                ArrayList<JsonElement> values = (ArrayList<JsonElement>) aggregation.get(entry);
-                JsonElement value = values.get(i);
-                String stringValue = getStringValueForJsonElement(value);
-                if (first) {
-                    valuesForInsert += stringValue;
-                    first = false;
-                } else {
-                    valuesForInsert += "," + stringValue;
-                } // if else
-            } // while
-            valuesForInsert += ")";
-        } // for
-        return valuesForInsert;
-    } // getValuesForInsert
-
-    private String getName() {
-        return "NGSIUtils.GenericAggregator";
+    public void setOrgName(String orgName) {
+        this.orgName = orgName;
     }
 
-    /**
-     * Gets fields for create.
-     *
-     * @return the fields for create
-     */
-    public String getFieldsForCreate() {
-        String fieldsForCreate = "(";
-        boolean first = true;
-        Iterator<String> it = aggregation.keySet().iterator();
-
-        while (it.hasNext()) {
-            if (first) {
-                fieldsForCreate += (String) it.next() + " text";
-                first = false;
-            } else {
-                fieldsForCreate += "," + (String) it.next() + " text";
-            } // if else
-        } // while
-
-        return fieldsForCreate + ")";
-    } // getFieldsForCreate
+    public String getPkgName(boolean enableLowercase) {
+        if (enableLowercase) {
+            return pkgName.toLowerCase();
+        } else {
+            return pkgName;
+        }
+    }
 
-    /**
-     * Gets fields for insert.
-     *
-     * @return the fields for insert
-     */
-    public String getFieldsForInsert() {
-        String fieldsForInsert = "(";
-        boolean first = true;
-        Iterator<String> it = aggregation.keySet().iterator();
-        while (it.hasNext()) {
-            if (first) {
-                fieldsForInsert += (String) it.next();
-                first = false;
-            } else {
-                fieldsForInsert += "," + (String) it.next();
-            } // if else
-        } // while
-        return fieldsForInsert + ")";
-    } // getFieldsForInsert
+    public void setPkgName(String pkgName) {
+        this.pkgName = pkgName;
+    }
+
+    public String getResName(boolean enableLowercase) {
+        if (enableLowercase) {
+            return resName.toLowerCase();
+        } else {
+            return resName;
+        }
+    }
+
+    public void setResName(String resName) {
+        this.resName = resName;
+    }
+
+    public boolean isAttrNativeTypes() {
+        return attrNativeTypes;
+    }
+
+    public void setAttrNativeTypes(boolean attrNativeTypes) {
+        this.attrNativeTypes = attrNativeTypes;
+    }
+
+    public boolean isEnableGrouping() {
```

Comment:
Maybe it is to be done in a next commit, but the style has somehow broken... Functions should have a JavaDoc description and the blocks an ending `// comment`.

Reply:
Fixed in [683ab07](https://github.com/telefonicaid/fiware-cygnus/commit/683ab076f020f01987369925c4e9091d083b7560)